Source/AwsCommonRuntimeKit/sdkutils/FileBasedConfiguration.swift

```diff
@@ -0,0 +1,203 @@
+        else {
+            return nil
+        }
+        return Section(rawValue: sectionPointer, fileBasedConfiguration: self)
+    }
+
+    deinit {
+        aws_profile_collection_release(rawValue)
+    }
+}
+
+
+
+
+extension FileBasedConfiguration {
+
+    /// Type of section in a config file
+    public enum SectionType {
+        case profile
+        case ssoSession
+    }
+
+    /// Represents a section in the FileBasedConfiguration
+    public class Section {
+        let rawValue: OpaquePointer
+        // Keep a reference of configuration to keep it alive
+        let fileBasedConfiguration: FileBasedConfiguration
+
+        init(rawValue: OpaquePointer, fileBasedConfiguration: FileBasedConfiguration) {
+            self.rawValue = rawValue
+            self.fileBasedConfiguration = fileBasedConfiguration
+        }
+
+        /// Returns a reference to the name of the provided profile
+        public var name: String {
+            String(awsString: aws_profile_get_name(rawValue))!
+        }
+
+        /// Retrieves a reference to a property with the specified name, if it exists, from a profile
+        /// - Parameters:
+        ///   - name: The name of property to retrieve
+        ///   - allocator: (Optional) default allocator to override
+        /// - Returns: A reference to a property with the specified name, if it exists, from a profile
+        public func getProperty(
+            name: String,
+            allocator: Allocator = defaultAllocator
+        ) -> FileBasedConfiguration.Section.Property? {
+            let nameAwsString = AWSString(name, allocator: allocator)
+            guard let propPointer = aws_profile_get_property(rawValue, nameAwsString.rawValue) else {
+                return nil
+            }
+            return FileBasedConfiguration.Section.Property(
+                rawValue: propPointer,
+                fileBasedConfiguration: fileBasedConfiguration)
+        }
+
+        /// Returns how many properties a section holds
+        public var propertyCount: Int {
+            aws_profile_get_property_count(rawValue)
+        }
+    }
+}
+
+extension FileBasedConfiguration.SectionType {
+    var rawValue: aws_profile_section_type {
+        switch self {
+        case .profile: return AWS_PROFILE_SECTION_TYPE_PROFILE
+        case .ssoSession: return AWS_PROFILE_SECTION_TYPE_SSO_SESSION
+        }
+    }
+}
+
+extension FileBasedConfiguration.Section {
+    /// Represents a section property in the file based configuration.
+    public class Property {
+        let rawValue: OpaquePointer
+        // Keep a reference of configuration to keep it alive
+        let fileBasedConfiguration: FileBasedConfiguration
+
+        init(rawValue: OpaquePointer, fileBasedConfiguration: FileBasedConfiguration) {
+            self.rawValue = rawValue
+            self.fileBasedConfiguration = fileBasedConfiguration
+        }
+
+        /// Returns the property's string value
+        public var value: String {
```

Comment:
What does this return if the property has sub properties?

Reply:
A string of all the sub properties, like for this config it returns "max_concurrent_requests = 20"